### DISCUSSION TITLE
Having trouble making the basic example work. Problems with SSR?

### DISCUSSION THREAD

**Discussion by zachdaniel** [react-hook-form/react-hook-form#5710] [Q&A] [answered]
I'm honestly having trouble getting react-hook-form to do...anything.
Does anyone have any experience with using react-hook-form along with server-side rendering? Could that be causing problems for me? I've copied the simple getting started example into my application, the one that watches a field and logs the watched value, but nothing seems to be happening. No rerenders on input, no required validation showing up. My application is pretty simple at the moment aside from the fact that it is doing SSR, which seems like a very likely candidate to be the cause here. Does `react-hook-form` work with server-rendered applications? If so, are there things I should know/patterns I should follow?

Thanks!

**Comment by zachdaniel** [comment]
I've also tried wrapping it in a `NoSSR` component that I copied from here: https://www.npmjs.com/package/react-no-ssr, and that didn't resolve the issue either, which leaves me wondering if maybe I've just somehow missed something really basic.

**Reply by bluebill1049** [reply]
I am using it with nextjs and have no problem, what's the error log?

**Comment by zachdaniel** [comment, marked as answer]
I've figured it out, and its totally on me :)

My SSR strategy involves having a `*_client.js` that finds the correct react component and renders it on the client. I have multiple `*_client.js` for various entrypoints that are meant to SSR a different set of components, and the client.js for each of them was being loaded, e.g `home_client.js` and `other_thing_client.js`. I'm not sure exactly what effect it had, but a lot of strange and mysterious symptoms went away once I fixed it.

EDIT:
I'm using elixir_react_render https://github.com/revelrylabs/elixir_react_render which is probably not something anyone else will end up encountering problems with, but I've added it here in case anyone might stumble here from google.
